src/pages/Mission.js

```diff
@@ -1,9 +1,25 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchMissions, selectMissions } from '../redux/mission/missionSlice';
 
-const Mission = () => (
-  <div>
-    This is the Mission page.
-  </div>
-);
+const Mission = () => {
+  const dispatch = useDispatch();
+  const missions = useSelector(selectMissions);
+
+  useEffect(() => {
+    dispatch(fetchMissions());
+  }, [dispatch]);
```

Comment:
The project requirement that states: `NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).`. Based on your implementation, whenever  changes a user navigates to a different link and revisit the mission link another dispatch action is fired.

To fix this;
- Kindly implement the useEffect hook in `App.js`.
- With this whenever the main app is loaded on the browser the data for the missions is fetched.
- And when the mission component is mounted, no dispatch action would be fired since the data is already fetched in the redux store.